packages/style-engine/class-wp-style-engine.php

```diff
@@ -236,6 +255,67 @@ public static function get_instance() {
 		return static::$instance;
 	}
 
+	/**
+	 * Taken from gutenberg_enqueue_block_support_styles()
+	 *
+	 * This function takes care of adding inline styles
+	 * in the proper place, depending on the theme in use.
+	 *
+	 * For block themes, it's loaded in the head.
+	 * For classic ones, it's loaded in the body
+	 * because the wp_head action  happens before
+	 * the render_block.
+	 *
+	 * @param callable $callable A user-defined callback function for a WordPress hook.
+	 * @param int      $priority To set the priority for the add_action.
+	 *
+	 * @see gutenberg_enqueue_block_support_styles()
+	 */
+	protected static function enqueue_registered_styles( $callable, $priority = 10 ) {
+		if ( ! $callable ) {
+			return;
+		}
+		$action_hook_name = 'wp_footer';
+		if ( wp_is_block_theme() ) {
+			$action_hook_name = 'wp_head';
+		}
+		add_action( 'wp_enqueue_scripts', $callable );
+		add_action(
+			$action_hook_name,
+			$callable,
+			$priority
+		);
+	}
+
+	/**
+	 * Fetches, processes and compiles stored styles, then renders them to the page.
+	 */
+	public static function enqueue_rendered_styles() {
+		foreach ( static::$stores as $store_key => $store_instance ) {
+			if ( ! $store_instance ) {
+				continue;
+			}
+
+			$css_rules = $store_instance->get_all_rules();
+
+			if ( empty( $css_rules ) ) {
+				continue;
+			}
+
+			$styles_output = '';
+
+			foreach ( $css_rules as $selector => $css_rule ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```

Comment:
Tiny nit: can we avoid the `phpcs:ignore` line by removing `$selector` from the `foreach`, or is it still needed for follow-ups?

```suggestion
			foreach ( $css_rules as $css_rule ) {
```

Reply:
Defo. Thanks. That was me trying to check my work in on a Friday afternoon 😄